homeassistant/components/sonos/media_player.py

```diff
@@ -1281,3 +1338,140 @@ def device_state_attributes(self):
             attributes[ATTR_QUEUE_POSITION] = self.queue_position
 
         return attributes
+
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Implement the websocket media browsing helper."""
+        if media_content_type in [None, "library"]:
+            return await self.hass.async_add_executor_job(
+                library_payload, self._media_library
+            )
+
+        payload = {
+            "search_type": media_content_type,
+            "idstring": media_content_id,
+            "full_album_art_uri": True,
+            "max_items": 0,
+        }
+        response = await self.hass.async_add_executor_job(
+            build_item_response, self._media_library, payload
+        )
+        if response is None:
+            raise BrowseError(
+                f"Media not found: {media_content_type} / {media_content_id}"
+            )
+        return response
+
+
+def build_item_response(media_library, payload):
+    """Create response payload for the provided media query."""
+    if payload["search_type"] == TYPE_ALBUM and payload["idstring"].startswith(
+        ("A:GENRE", "A:COMPOSER")
+    ):
+        payload["idstring"] = "A:ALBUMARTIST/" + "/".join(
+            payload["idstring"].split("/")[2:]
+        )
+
+    media = media_library.browse_by_idstring(**payload)
+
+    if media is None:
+        return
+
+    thumbnail = None
+    title = None
+
+    # Fetch album info for titles and thumbnails
+    # Can't be extracted from track info
+    if (
+        payload["search_type"] == TYPE_ALBUM
+        and media[0].item_class == "object.item.audioItem.musicTrack"
+    ):
+        for item in media_library.browse_by_idstring(
+            TYPE_ALBUM_ARTIST,
+            "/".join(payload["idstring"].split("/")[0:]),
```

Comment:
This line doesn't do anything? You're splitting it by `/`, then making a copy that starts at index 0 till the end, and then joining it again using `/`
```suggestion
            payload["idstring"],
```